### PR TITLE
Reduce LTX2.3 CFG input prep overhead

### PR DESCRIPTION
**Pull request by TaffyOfficial**
## Purpose

Reduce LTX-2.3 single-rank CFG input preparation overhead.

The previous path duplicated video and audio latents with `torch.cat` while they were still fp32, then cast the doubled tensors to the connector prompt dtype before calling the transformer. This PR casts the video/audio latents to the connector prompt dtype first, then performs the CFG duplication on the already-cast tensors. The transformer still receives the same batch shape and dtype, but the input-prep path avoids building a doubled fp32 tensor only to immediately copy/cast it.

Scope is intentionally limited to this dtype-before-cat optimization and its pipeline mock coverage. The attention-mask metadata/scalar-sync experiment is not included because it did not preserve LTX-2.3 output accuracy.

## Test Plan

- `pytest -q tests/diffusion/models/ltx2/test_ltx2_3_pipeline.py`
- CUDA micro-profile of the old and new CFG input-prep tensor sequence using LTX-2.3 512x384, 25-frame latent-scale shapes.

## Test Result

- Remote `venv-vllm-0.23`: `pytest -q tests/diffusion/models/ltx2/test_ltx2_3_pipeline.py` passed with `34 passed`.
- The new coverage exercises the single-rank CFG forward path with fp32 video/audio latents and verifies that bf16 video/audio hidden states reach the transformer after CFG duplication.
- CUDA operator micro-profile on L20X, 20 input-prep iterations per repeat:
  - old path target ops (`aten::cat` + cast `aten::copy_`): `13.54 ms` device time across the profiled window.
  - new path target ops (`aten::cat` + cast `aten::copy_`): `9.83 ms` device time across the profiled window (`-27.5%`).
  - alternating CUDA-event timing over 12 rounds: old path `0.6480 ms` mean per 20 input-prep iterations; new path `0.6330 ms` mean (`+2.3%` local input-prep speedup).

### CODE REVIEW

**Review by hsliuustc0106 — COMMENTED** [review]
3 comments above
